src/main/java/de/retest/recheck/report/TestReportFilter.java

```diff
@@ -146,7 +147,12 @@ StateDifference filter( final StateDifference stateDiff ) {
 	}
 
 	Optional<InsertedDeletedElementDifference> filter( final InsertedDeletedElementDifference insertedDeletedDiff ) {
-		return filter.matches( insertedDeletedDiff.getInsertedOrDeletedElement() ) //
+		if ( insertedDeletedDiff.isInserted() ) {
+			return filter.matches( insertedDeletedDiff.getInsertedOrDeletedElement(), ChangeType.Inserted ) //
```

Comment:
But the same is also true for any other impl. It's no difference here...?

Reply:
I just want to point out that we should either document the method call order, so that potential implementations know that 

1. no other method is called in case of inserted/deleted, 
2. or change this implementation to check first if `matches(Element)` is true in which case `matches(Element, ChangeType)` is never called. This would be somewhat in line with `matches(Element, AttributeDifference)`.

Reply:
Will call `matches(Element)` first.